newDB.sql

```diff
@@ -1,97 +1,110 @@
+  updated_at timestamp without time zone,
+  check(updated_at > created_at)
 );
 
-create table Customer (
-  CustomerID int not null primary key,
-  IFS_Code varchar(255) not null references Branch(IFS_Code),
-  F_Name varchar(255) not null,
-  L_Name varchar(255) not null,
-  Login_ID varchar(255) not null,
-  Acc_Pass varchar(255) not null,
-  Trans_Pass varchar(255) not null,
-  Acc_Status varchar(255) not null,
-  City varchar(255) not null,
-  State varchar(255) not null,
-  Country varchar(255) not null,
-  Acc_Open_Date date not null,
-  Last_login date not null
-);
+create index branch_idx on Customers(branch_id);
 
 create table Accounts(
-  Account_No varchar(255) not null primary key,
-  customer_ID int not null references Customer(CustomerID),
-  acc_status varchar(255) not null,
-  acc_open_date date not null,
-  account_type varchar(255) not null,
-  account_balance decimal not null
+  id bigserial primary key,
+  number bigint not null unique,
+  atm_pin int not null default 1234,
+  customer_id bigint not null references Customers(id),
+  status varchar(255) not null,
+  type text not null,
+  balance float not null,
+  created_at timestamp without time zone not null default now(),
+  updated_at timestamp without time zone,
+  check(created_at < updated_at)
 );
 
-create table Employee (
-  Emp_ID int not null primary key,
-  Emp_Name varchar(255) not null,
-  Login_ID varchar(255) not null,
-  password varchar(25) not null,
-  Email_ID varchar(255) not null,
-  Contact_No varchar(25) not null,
-  Create_Date date not null,
-  Last_Login date not null
-);
+create index accounts_customer_id_index on Accounts(customer_id);
 
-create table Loan (
-  Loan_ID int not null primary key,
-  Loan_Type varchar(25) not null,
-  Loan_Amount varchar(25) not null,
-  Customer_ID int not null references Customer(CustomerID),
-  Interest float not null,
-  Start_Date date not null
+create table Loans (
+  id bigserial primary key,
+  customer_id bigint not null references Customers(id),
+  type text not null,
+  amount float not null,
+  interest float not null,
+  period float not null,
+  balance float not null,
+  created_at timestamp without time zone null default now(),
+  monthly_emi float not null,
+  monthly_emi_remaining int not null,
+  monthly_emi_date date not null,
+  updated_at timestamp without time zone,
+  check (balance <= amount),
+  check (updated_at > created_at)
+  check (monthly_emi_remaining < period)
 );
 
-create table Loan_Payment (
-  Payment_ID int not null primary key,
-  Date date not null,
-  Paid_Amount float not null,
-  Principle_Amount float not null,
-  Balance float not null
+create index loans_customer_id_index on Loans(customer_id);
+
+--check for prepaid??
+create table Loan_Payments (
+  id bigserial primary key,
+  loan_id bigint not null references Loans(id),
+  payment_id varchar(255) not null unique,
+  date timestamp without time zone not null,
```

Comment:
Should avoid using reserved keywords for column names

Reply:
Okay sir will update it